### PR TITLE
bpo-38823: Clean up refleaks in _ast initialization.

### PR DESCRIPTION
**Pull request by brandtbucher**
https://bugs.python.org/issue38823

### CODE REVIEW

**Review by corona10 — APPROVED** [review]
PR looks good to me.
The specific change is that error handling is done by the goto statement
and this decision is looking good to me for this situation.

@vstinner 
Please take a look at @brandtbucher 's PR as the core developer.
Thank you for understanding.

File: Parser/asdl_c.py [hunk 1]
```diff
@@ -998,17 +998,20 @@ def visitModule(self, mod):
         self.emit("if (!init_types()) return NULL;", 1)
         self.emit('m = PyState_FindModule(&_astmodule);', 1)
         self.emit("if (!m) return NULL;", 1)
+        self.emit('if (PyModule_AddObject(m, "AST", astmodulestate_global->AST_type) < 0) goto error;', 1)
```
Comment (vstinner) [inline Parser/asdl_c.py]: Please add braces, see PEP 7.

```suggestion
        self.emit('if (PyModule_AddObject(m, "AST", astmodulestate_global->AST_type) < 0) { goto error; }', 1)
```

Same for changes below.

**Comment by csabella** [comment]
@brandtbucher, please address the review comments.  Thanks!

**Comment by brandtbucher** [comment]
Sorry @vstinner, I didn't realize that PEP 7 applied to generated code. Fixed!

**Review by shihai1991 — APPROVED** [review]
Thanks, LGTM.

**Comment by vstinner** [comment]
I don't see the need to backport such change. Errors in _ast init function are very unlikely. I prefer to reduce the number of backported changes to stable branches to reduce the risk of regression.

**Comment by brandtbucher** [comment]
Thanks!

**Pull request merged**